packages/@aws-cdk/aws-iot-actions/lib/cloudwatch-alarm-action.ts

```diff
@@ -0,0 +1,49 @@
+import * as cloudwatch from '@aws-cdk/aws-cloudwatch';
+import * as iam from '@aws-cdk/aws-iam';
+import * as iot from '@aws-cdk/aws-iot';
+import { CommonActionProps } from './common-action-props';
+import { singletonActionRole } from './private/role';
+
+/**
+ * Configuration properties of an action for CloudWatch alarm.
+ */
+export interface CloudWatchAlarmActionProps extends CommonActionProps {
+  /**
+   * The reason for the alarm change.
+   */
+  readonly stateReason: string;
```

Comment:
1. Let's just call this `reason`.
2. Seems like we can probably default this to something sensible...? I don't think we need to make this required.

Reply:
Fixed.